### PR TITLE
Do not recalculate shipping if PS_ORDER_RECALCULATE_SHIPPING is false

### LINKED ISSUE #20744: Recalculate shipping costs option not working when set to No
#### Describe the bug

When setting the "Recalculate shipping costs after editing the order" option to No, the shipping cost are still recalculated when editing the order in BO when the price of the order exceeds the carrier's "limit".

#### Expected behavior

The shipping cost should not be updated if option is set to No.

#### Steps to Reproduce

0. Have a carrier set up with a fee below a certain amount, free above it
1. Go to BO > Shop Parameters > Order Settings
2. Set "Recalculate shipping costs after editing the order" to No
3. On FO, make an order with this carrier, under the limit amount (50€ in my screenrecord)
4. On BO, edit the order to make it exceed the limit

**Screenshots**

On this screenrecord : The carrier is set to be at 8€40 under 50€, 2€40 over 50€. When editing the order in BO the shipping cost should not be updated when the total of the order is over 50€, but it is.

https://drive.google.com/file/d/13R8qxiwqy0xfjpdUmrNNiKVdP3HkXzj7/view?usp=sharing

#### Additional information

* PrestaShop version: 1.7.7
* PHP version: N/A

### PR DESCRIPTION
**Pull request by atomiix**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | When an order was modified in the BO, shipping fees where recalculated whether or not `PS_ORDER_RECALCULATE_SHIPPING` was set to false.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #20744
| How to test?  | See #20744

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20798)

### CODE REVIEW

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Please use `PrestaShop\PrestaShop\Adapter\Configuration` rather than `Configuration` from legacy

File: src/Adapter/Order/OrderAmountUpdater.php [hunk 1]
```diff
@@ -107,13 +108,29 @@ public function update(
...
+        $totalShippingTaxExcluded = $order->total_shipping_tax_excl;
+
         $order->total_shipping = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
         $order->total_shipping_tax_excl = $cart->getOrderTotal(false, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
         $order->total_shipping_tax_incl = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
 
-        $order->total_wrapping = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
-        $order->total_wrapping_tax_excl = abs($cart->getOrderTotal(false, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
-        $order->total_wrapping_tax_incl = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
+        if (!Configuration::get('PS_ORDER_RECALCULATE_SHIPPING')) {
```
Comment (Progi1984) [inline src/Adapter/Order/OrderAmountUpdater.php]:
```suggestion
        if (!$this->configuration->get('PS_ORDER_RECALCULATE_SHIPPING')) {
```

File: src/Adapter/Order/OrderAmountUpdater.php [hunk 2]
```diff
@@ -36,6 +36,7 @@
 use Order;
 use OrderCarrier;
 use OrderCartRule;
+use PrestaShop\PrestaShop\Adapter\Configuration;
```
Comment (Progi1984) [inline src/Adapter/Order/OrderAmountUpdater.php]:
```suggestion
```

File: src/Adapter/Order/OrderAmountUpdater.php (hunk 1 above)
Comment (atomiix) [reply src/Adapter/Order/OrderAmountUpdater.php]: 🤦‍♂️

**Review by Progi1984 — APPROVED** [review]

File: src/Adapter/Order/OrderAmountUpdater.php [hunk 3]
```diff
@@ -107,13 +107,29 @@ public function update(
...
+
         $order->total_shipping = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
         $order->total_shipping_tax_excl = $cart->getOrderTotal(false, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
         $order->total_shipping_tax_incl = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
 
-        $order->total_wrapping = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
-        $order->total_wrapping_tax_excl = abs($cart->getOrderTotal(false, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
-        $order->total_wrapping_tax_incl = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
+        if (!$this->configuration->get('PS_ORDER_RECALCULATE_SHIPPING')) {
+            $shippingDiffTaxIncluded = $order->total_shipping_tax_incl - $totalShippingTaxIncluded;
```
Comment (sowbiba) [inline src/Adapter/Order/OrderAmountUpdater.php:122]: but $totalShippingTaxIncluded == $order->total_shipping_tax_incl

So $totalShippingTaxIncluded == 0

Cant we just not modify total_paid, total_paid_tax_incl and total_paid_tax_excl ?

Comment (atomiix) [reply src/Adapter/Order/OrderAmountUpdater.php:122]: @sowbiba I'm not sure to understand but, `$order->total_shipping_tax_incl` is not 0 if shipping wasn't free 🤔 
Also, `total_paid`, `total_paid_tax_incl` & `total_paid_tax_excl` should be the result of shipping + products total (+ fees) so we cannot update one without updating the other.

Comment (sowbiba) [reply src/Adapter/Order/OrderAmountUpdater.php:122]: On line 114, we have
$totalShippingTaxIncluded = $order->total_shipping_tax_incl;
Line 122
$shippingDiffTaxIncluded = $order->total_shipping_tax_incl - $totalShippingTaxIncluded;

So $shippingDiffTaxIncluded == 0

Comment (sowbiba) [reply src/Adapter/Order/OrderAmountUpdater.php:122]: Ok, you are right, I just ignored the 3 line between these statements :D :D

**Review by sowbiba — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by ngodefroy** [comment]
TEST OK:
Before Fix:
We were able to modify the shipping price even if the "recalculate shipping costs after editing order" option was set.
https://drive.google.com/file/d/18SL9U1evNe2Ju-qYS8_H-E4L2Ki9SdtH/view


After Fix:
When the option is set 
https://drive.google.com/file/d/1DBvA0uekycqYelGeiNNEf6-XDUwy8U95/view

Checked that no regression was found disabling the "recalculate shipping costs after editing order" option: The price is well updated, everything correct. 

Thank you @sowbiba and @atomiix

**Comment by PierreRambaud** [comment]
Thanks  @atomiix

**Pull request merged**